test/physics/em/UrbanMsc.test.cc

```diff
@@ -285,24 +285,22 @@ TEST_F(UrbanMscTest, msc_scattering)
         angle.push_back(sample_result.direction[0]);
     }
 
-    const double expected_fstep[] = {0.0027916899999997,
-                                     0.134631648532277,
-                                     0.0376414,
-                                     0.0357277834605262,
-                                     0.00111391683751815,
-                                     0.000112053935348643,
-                                     0.000286789820693628,
-                                     1.17373195131411e-05};
-
-    const double expected_angle[] = {0.000314741326035635,
-                                     0.738624667826603,
-                                     -0.145610123961716,
-                                     -0.657415795200799,
-                                     0.119014628205848,
-                                     -0.216102964881347,
-                                     0.793645871128744,
-                                     -0.98020130119347};
-
+    static const double expected_fstep[] = {0.0027916899999997,
```

Comment:
Now results are quite different, which infers that tweaking codes in this MR introduced something logically different from original codes (which I cannot identify easily now, but we need to make sure that something is not wrongly changed - unless the random sequence is changed).

Reply:
I made a note in the final commit, but this is strictly due to the ordering of the RNG sampling (which I think uses fewer temporaries now). Moving the polar angle sampling *before* the azimuthal angle restores the original results.

Reply:
Make a sense if the sampling order has changed.